compiler/index.js

```diff
@@ -1,7 +1,63 @@
-var handlebars = require('handlebars')
+var Handlebars = require('handlebars');
+var walkSync = require('walk-sync');
+var fs = require('fs');
+var path = require('path');
 
 function Compiler(rootPath) {
-  this.root = rootPath;
+  this.rootPath = rootPath;
 }
 
+
+Compiler.prototype.collectInputFilePaths = function(dirPath, type) {
+  var inputPath = path.join(this.rootPath, dirPath);
+  var filePaths = walkSync(inputPath).filter(function(filePath) {
+    return filePath.substr(filePath.length - type.length) === type;
+  });
+
+  return filePaths;
+};
+
+Compiler.prototype.registerWithHandlebars = function(filePath) {
```

Comment:
I think this might be cleaner to split into two functions:

```javascript
Compiler.prototype.registerHelper = function(filePath) {
  var inputPath = path.join(this.rootPath + filePath);

  var helperFunction = require(inputPath);
  Handlebars.registerHelper(name, helperFunction);
};

Compiler.prototype.registerPartial = function(filePath) {
  var inputPath = path.join(this.rootPath + filePath);
  var name = path.basename(inputPath, fileType);

  var partialString = fs.readFileSync(inputPath, 'utf8');
  Handlebars.registerPartial(name, partialString);
};
```

Removing the conditionals, also means less setup and makes it somewhat easier to read to me.  

What do you think?